### DISCUSSION TITLE
Nuxt Bridge for production How good is this idea?

### DISCUSSION THREAD

**Discussion by nestle49** [nuxt/nuxt#17220] [Questions]
Nuxt Bridge for production How good is this idea?

**Comment by mirumirumi** [comment]
I don't think it is very good.
I just finished an application with Nuxt 3 from scratch and it is quite nonfree and buggy. (Nuxt 3 itself is fantastic! ✨)

It would be extremely difficult to run it, including migration.

Maybe you could consider a different approach? :)

**Reply by nestle49** [reply]
Did you use nuxt-bridge or nuxt3?
Can you more talk about difficulties you encountered?

We use Nuxt2, now finished migrate from vue class component to composition api

We don’t know yet what would be the best to do - wait for the stable release nuxt3 and switch to the bridge or migrate to Vite (ssr) + Vue3?

**Reply by mirumirumi** [reply]
I have not used Nuxt Bridge, but I have used Nuxt 3 there.
The main difficulties are as follows:

- Page does not move to top after page moving (Nuxt 3 does not yet have something like scrollToTop)
- Difficult to determine whether it is executed on the server or the client
  - If you make the wrong decision, you will get hydration errors; ClientOnly, etc. are important
- Deployment approach in SSG
  - Nuxt 3 has not yet finished all the design and implementation regarding rendering modes ([see](https://v3.nuxtjs.org/guide/concepts/rendering#coming-in-nuxt-3))

However if you're all done migrating to the Composition API and you don't intend to use SSG (and it looks like you do) Nuxt 3 may already be an attractive way! :)

**Reply by nestle49** [reply]
Thanks, these are little things that we do not use

I would like to hear opinion from anyone who used Bridge in production
